common/ast/src/parser/rule/expr.rs

```diff
@@ -581,7 +581,13 @@ pub fn type_name(i: Input) -> IResult<TypeName> {
     let ty_float = value(TypeName::Float, rule! { FLOAT });
     let ty_double = value(TypeName::Double, rule! { DOUBLE });
     let ty_date = value(TypeName::Date, rule! { DATE });
-    let ty_datetime = value(TypeName::DateTime, rule! { DATETIME });
+    let ty_datetime = map(
+        rule! { DATETIME ~ ("(" ~ #literal_u64 ~ ")")? },
```

Comment:
I usually find it clearer to add space in between parenthesis.

```suggestion
        rule! { DATETIME ~ ( "(" ~ #literal_u64 ~ ")" )? },
```